common/source/docs/common-scripting-parameters.rst

```diff
@@ -0,0 +1,83 @@
+
+This adds two parameters, AERO_TRICK_ID and AERO_RPT_COUNT, with indexes 1 and 2, and initial values of 0 to that table. Other concurrently running scripts could also add parameters as long as they do not overlap index numbers.
+
+.. note:: if there are other con-currently running scripts that will try to access these parameters, be sure to delay their first updates until the script adding the params has time to complete the setup process since each script is run independently in its own thread
+
+These may now be accessed just like any other parameter. ::
+
+    local TRICK_ID = Parameter()
+    TRICK_ID:init('AERO_TRICK_ID')
+    local trick_id = TRICK_ID:get()
+
+.. note:: all scripting defined parameters are of type FLOAT, ie floating point numbers
+
+.. _how-params-work:
+
+How Parameters Work in ArduPilot
+================================
+
+Parameters are defined in the firmware. Each parameter has a name, a default value, and value type(boolean,float, integer 8,etc).
+Upon boot up, the firmware creates RAM variables for each parameter and initializes them with a value. That value is the default if the parameter has never been changed by a user, or a value that has been stored in EEPROM of the autopilot, if it has been changed from default. If the parameters have been "reset" by the user (see :ref:`common-parameter-reset`) then the EEPROM storage for changes is blank and all default values are used.
+
+The string names of the parameters are only used by the Ground Control Stations and LUA scripts. Access from within the code is directly to addresses of the initialized parameter variables in RAM while executing, not by name strings.
+
+LUA created parameters exist in the same manner as firmware created parameters. User/Script changes are stored in non-volatile memory. But their names only exist for a lookup if a "param:add_table" and "param:add_param" has been executed after bootup by a script. Then they will appear in the GCS parameter list and can be accessed by name from LUA scripts. Without this, the data would still exist in non-volatile memory, but is invisible and non-existent for all intents and purposes.
+
+This also means that parameters created by one script, which are modified by a user or script, are still in storage . If that script is removed, another script with different parameters is run, then removed, and the original script re-run, those changed parameters will re-appear in the GCS with their last values and can be accessed again by the script. Resetting all the parameters to defaults will erase those changes, just like the firmware created parameters.
+
+From LUA scripts you can access any parameter (created by firmware or by LUA script) by name. But that is slow since the LUA interpreter must do a long string name search each time. Instead using a Parameter() object and initializing it, gives the address pointers to that named parameter's location for direct accessing it quickly.
```

Comment:
```suggestion
From LUA scripts you can access any parameter (created by firmware or by LUA script) by name. However, that is slow since the LUA interpreter must do a long string name search each time. Instead, using a Parameter() object and initializing it, gives the address pointers to that named parameter's location for directly accessing it quickly. See "Accessing/Adding Parameters via Scripts" above.
```